policy/modules/services/dovecot.if

```diff
@@ -61,6 +61,68 @@ interface(`dovecot_domtrans_deliver',`
 	domtrans_pattern($1, dovecot_deliver_exec_t, dovecot_deliver_t)
 ')
 
+########################################
+## <summary>
+##	Read and write dovecot unix
+##	stream sockets.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`dovecot_rw_stream_sockets',`
+	gen_require(`
+		type dovecot_t;
+	')
+
+	allow $1 dovecot_t:unix_stream_socket { connectto rw_socket_perms };
```

Comment:
the `rw_socket_perms` vs `connectto` seems odd. Is it receiving a fd for a socket from dovecot?

Reply:
> the `rw_socket_perms` vs `connectto` seems odd. Is it receiving a fd for a socket from dovecot?

Thanks for raising this for review again. Looks like I got this access mixed up actually after another thorough look, so `rw_socket_perms` isn't needed at all here.:
```
time->Wed Feb  3 11:16:14 2021
type=PROCTITLE msg=audit(1612368974.831:29966): proctitle=2F7573722F6C6962657865632F646F7665636F742F64656C69766572002D6600746F6E79746865736C6179657240676D61696C2E636F6D002D64006D6540636F6E636F72642E7368
type=PATH msg=audit(1612368974.831:29966): item=0 name="/run/dovecot/stats-writer" inode=1652 dev=00:11 mode=0140660 ouid=0 ogid=5000 rdev=00:00 obj=system_u:object_r:dovecot_runtime_t:s0 nametype=NORMAL cap_fp=0 cap_fi=0 cap_fe=0 cap_fver=0 cap_frootid=0
type=CWD msg=audit(1612368974.831:29966): cwd="/var/spool/postfix"
type=SOCKADDR msg=audit(1612368974.831:29966): saddr=01002F72756E2F646F7665636F742F73746174732D7772697465720000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
type=SYSCALL msg=audit(1612368974.831:29966): arch=c000003e syscall=42 success=yes exit=0 a0=7 a1=779e570b52d8 a2=6e a3=6f9e24fa2aa0 items=1 ppid=2010426 pid=2010427 auid=4294967295 uid=5000 gid=5000 euid=5000 suid=5000 fsuid=5000 egid=5000 sgid=5000 fsgid=5000 tty=(none) ses=4294967295 comm="deliver" exe="/usr/libexec/dovecot/dovecot-lda" subj=system_u:system_r:postfix_pipe_t:s0 key=(null)
type=AVC msg=audit(1612368974.831:29966): avc:  denied  { connectto } for  pid=2010427 comm="deliver" path="/run/dovecot/stats-writer" scontext=system_u:system_r:postfix_pipe_t:s0 tcontext=system_u:system_r:dovecot_t:s0 tclass=unix_stream_socket permissive=1
type=AVC msg=audit(1612368974.831:29966): avc:  denied  { write } for  pid=2010427 comm="deliver" name="stats-writer" dev="tmpfs" ino=1652 scontext=system_u:system_r:postfix_pipe_t:s0 tcontext=system_u:object_r:dovecot_runtime_t:s0 tclass=sock_file permissive=1
```